### PR TITLE
Entangling thrown weapons - nets and bolas

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Entangling thrown weapons - net and bolas"```

This has the same changes as #32068 as it is built upon it.

#### Purpose of change
Wanted a way to slow down a wild horse for me to tame it, before it ran away.
This led me to think of bolas and nets to tangle up creatures.
A Lasso could be similar.

#### Describe the solution
Added a TANGLE flag to weapons, if an ammo or a thrown weapon has this flag, and the projectile hits a creature, it will tangle them up, and the item will not be retrievable until the creature breaks free.
For a tamed creature it will tie them up as if you had tied them with a rope, and the tied_item attached to monster class, will then be the bolas or net - that you can untie and retrieve. ( not the most humane way to keep them in one place, admittedly )

For a creature that isnt tame, it will try and break free, sometimes breaking the item in the process, otherwise it will be tied to the spot for a while, useful to get away from creatures, or to keep them still while you attack them from a distance.

For an NPC or player, it will apply downed effect and stunned effect briefly, humans are better at untying themselves, but the effects simulate the staggering and stumbling as they do so.

#### Describe alternatives you've considered
N/A

#### Additional context
I want to look into further work in this area, with ropes being used as lassos, and /or tying up / capturing creatures/NPCs, this is a first step in that area - not that I want to do anything nefarious to tied up NPCs mind you, just that capturing in a non-lethal way could be useful at some point.

Guns that have ammo with this flag could be made - a net gun? maybe? or other more exotic things. as a non-lethal solution.

I don't often do combat-related PRs as I dont like balancing stuff and I don't trust my judgement much in that area, this was just a side-effect of wanting to capture wild horses, so I went with it.
Welcome feedback from those who do know how to balance combat mechanics.

### CODE REVIEW

**Comment by nexusmrsep** [comment]
Certainly opens up the road to traps of similar concept.

File: data/json/items/melee.json [hunk 1]
```diff
@@ -49,6 +49,38 @@
     "ammo_type": "thrown",
     "flags": [ "NOGIB" ]
   },
+  {
+    "id": "bolas",
+    "type": "AMMO",
+    "category": "weapons",
+    "name": "bolas",
```
Comment (Night-Pryanik) [inline data/json/items/melee.json]: Plural name needed.

File: src/monster.cpp [hunk 2]
```diff
@@ -1480,6 +1483,45 @@ bool monster::move_effects( bool )
...
+            bool immediate_break = type->in_species( FISH ) || type->in_species( MOLLUSK ) ||
+                                   type->in_species( ROBOT ) || type->bodytype == "snake" || type->bodytype == "blob";
+            if( !immediate_break && rng( 0, 900 ) > type->melee_dice * type->melee_sides * 1.5 ) {
+                if( u_see_me ) {
+                    add_msg( _( "The %s struggles to break free of its bonds." ), name() );
+                }
+            } else if( immediate_break ) {
+                remove_effect( effect_tied );
+                if( tied_item ) {
+                    add_msg( _( "The %s easily slips out of its bonds." ), name() );
```
Comment (BevapDin) [inline src/monster.cpp]: There is no `u_see_me` check for this message like it is for the other two messages.

File: src/monster.cpp [hunk 3]
```diff
@@ -1480,6 +1483,45 @@ bool monster::move_effects( bool )
...
+                    g->m.add_item_or_charges( pos(), *tied_item );
+                    tied_item = cata::nullopt;
+                }
+            } else {
+                bool broken = true;
+                if( tied_item ) {
+                    broken = rng( type->melee_dice * type->melee_sides, std::min( 10000,
```
Comment (BevapDin) [inline src/monster.cpp]:
```suggestion
                    const bool broken = rng( type->melee_dice * type->melee_sides, std::min( 10000,
```
No need to declare it outside of this block.

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by davidpwbrown** [comment]
jenkins rebuild

File: src/creature.cpp [hunk 4]
```diff
@@ -661,6 +662,26 @@ void Creature::deal_projectile_attack( Creature *source, dealt_projectile_attack
...
+                z->tied_item = proj.get_drop();
+            } else {
+                add_msg( m_debug, "projectile with TANGLE effect, but no drop item specified" );
+            }
+        } else if( n && !is_immune_effect( effect_downed ) ) {
+            // no tied up effect for people yet, just down them and stun them, its close enough to the desired effect.
+            // we can assume a person knows how to untangle their legs eventually and not panic like an animal.
+            add_effect( effect_downed, 1_turns );
+            // stunned to simulate staggering around and stumbling trying to get the entangled thing off of them.
+            add_effect( effect_stunned, 1_turns * rng( 3, 8 ) );
```
Comment (BevapDin) [inline src/creature.cpp]:
```suggestion
            add_effect( effect_stunned, rng( 3_turns, 8_turns ) );
```

File: src/monster.cpp [hunk 5]
```diff
@@ -1480,6 +1483,46 @@ bool monster::move_effects( bool )
 
     bool u_see_me = g->u.sees( *this );
     if( has_effect( effect_tied ) ) {
+        // friendly pet, will stay tied down and obey.
+        if( friendly == -1 ) {
+            return false;
+        } else {
```
Comment (BevapDin) [inline src/monster.cpp]:
```suggestion
        }
```
And than you can remove one level of indentation below.

**Pull request merged**